### PR TITLE
Make OW4 an OpenID Connect Provider

### PR DESCRIPTION
**Pull request by sklirg**
## What kind of a pull request is this?

- [x] QA / Code Review

## Code Checklist

- [x] The code follows dotkom code style 
- [x] The code passes the defined tests
- [x] The code is ready to be merged

## (Possible) Breaking Changes

- [x] The changes are backwards compatible
- [x] These changes requires changes to configuration in production
    - [ ] I have applied the required changes in production
    - [x] I cannot apply the required changes in production before this is deployed.

## Description of changes

Makes OW4 an OpenID Connect provider so that we can create clients using OIDC rather than pure OAuth2.

The change required in production is to run `python manage.py creatersatoken` to create an RSA token for the OIDC server to provide.

### CODE REVIEW

File: templates/oidc_provider/authorize.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            {% endfor %}
+            </ul>
+            <br />
+            <div id="sso-control-buttons">
+                <input type="submit" class="btn btn-large btn-success" name="allow" value="Godta" />
+                <input type="submit" class="btn btn-large btn-danger pull-right" value="Avslå" />
+            </div>
+        </form>
+    </div>
+    <p class="text-center"><small><i class="glyphicon glyphicon-lock"></i> Ekstern autentisering benytter OAuth 2.0 gjennom TLS</small></p>
```
Comment (duvholt) [inline templates/oidc_provider/authorize.html]: Copypaste? 😄

Comment (sklirg) [inline templates/oidc_provider/authorize.html]: Yes, is it not relevant anymore? Could probably reword it to say "using OAuth2 and OpenID".

Comment (sklirg) [inline templates/oidc_provider/authorize.html]: Also: I'm not sure I'd treat that as a blocking issue (and therefore a "request for change"), but rather a small refactoring task. This feature is interesting to get merged ASAP and such a small issue can be handled in a new issue (labeled easy!) instead IMO ;-)

Comment (duvholt) [inline templates/oidc_provider/authorize.html]: Didn't consider that OIDC is built on OAuth2 and therefore the view can also be used for OAuth2. 

Regarding "request for change": IMO it's better to fix quick issues like this early, otherwise they rarely get done. In this case it wasn't really an issue, but I thought it was supposed to say OpenID Connect instead of OAuth2.
